content/en/docs/concepts/instrumenting-library.md

```diff
@@ -0,0 +1,272 @@
+---
+title: "Instrumenting libraries"
+weight: 40
+---
+
+OpenTelemetry already provides [auto-instrumentations](./instrumenting.md#automatic-instrumentation)
+for many libraries, which is typically done through library hooks or monkey-patching library code.
+
+Native library instrumentation with OpenTelemetry provides better observability and developer experience for users, removing the need for libraries to expose and document hooks:
+
+- custom logging hooks can be replaced by common and easy to use OpenTelemetry APIs, users will only interact with OpenTelemetry
+- telemetry (traces, logs, metrics in future) from library and application code are correlated and coherent
+- common conventions allow users to get similar and consistent telemetry within same technology and across languages
+
+TODO/Open question: infra instrumentation

```

Comment:
Perhaps something like, "In the future, OpenTelemetry context will allow for easy correlation between infrastructure and application metrics, logs, and traces"

Reply:
yeah, I meant here that infra pieces (e.g. serverless environments, API gateways, etc) also could (and should) be instrumented, but I'm not currently ready to cover them in this article. I'll remove TODO for now and we can add them later in this or another article.